### ISSUE TITLE
[1.8.16 regression] multiple use of page label

### ISSUE THREAD

**Issue by rolegic** [doxygen/doxygen#7471]
For many years I was able to split documentation over several files, with cross-references:

file1.txt
```
/*!
@page my_doc        My Documentation

@ref item1      
@ref item2

@section item1      Item 1

First part.
*/
```

file2.txt
```
/*!
@page my_doc
@section item2      Item 2

Second part.
*/
```
→ Generation is Ok with one html page: `my_doc.html`

→ In the documentation, `page` is defined as follow `\page <name> (title)` ; in my understanding:
- on first use:  `<name>` and  `title` are needed to define the page
- on next use: `<name>` is only needed to reference/continue the page 

---
Since 1.8.16 (and still with 1.8.17) I have the following warning:
```
file2.txt:2: warning: multiple use of page label 'my_doc', (other occurrence:file1.txt, line: 3)
```
→ In this new use-case, what is the purpose of the optional `(title)` ?

- Why such a change?
- I use this "documentation split" for all of my manuals...
  - Do I need to rework them all?
  - And how? Which alternative/workaround do you propose?

**Comment by mosra** [comment]
Came here a minute ago to report the same, heh :)

In my case I have files named e.g. `building.dox`, with `@page building` inside, and it triggers the same warning. I suspect it's using the filename as page name and then complains the same-named page is used again?

> doc/building.dox:29: warning: multiple use of page label 'building', (other occurrence: <building>, line: 1)

Can it just ... not? I don't want to rename all my files to `building_.dox` etc. to prevent this warning -- I'd like to preserve the decade of Git history I have on some of those files. It's annoying as I want the output to be warning-free in order to discover new regressions in my written docs.

**Comment by hemberger** [comment]
I'm encountering the same issue. I can't find anything else in doxygen that would allow comments from multiple files to be displayed on the same page. A workaround or fix that doesn't trigger a warning would be much appreciated.

**Comment by teju85** [comment]
I too have been using the same approach to split the documentation comments inside source code. Even with version `1.8.20`, this still seems to be the issue?

I did try the `@xrefitem` as follows, but it seems to eat up all the contents and show nothing!?

```cpp
/**
 * @page Topic Some topic
 * All functions under Topic
 */

/**
 * @xrefitem Topic "" ""
 * @section Topic_subtopic
 * This is a sub-topic
 */
void some_function(...) {
}
```

It ends up creating a bunch of cross-referenced items in the `Topic.html` page which lead to the usual function documentations!

Atleast on v1.8.16, I could see a Topic.html file created with a section called `Topic_subtopic` under it!

**Comment by albert-github** [comment]
@teju85 I tried to reproduce your problem but neither with version 1.8.16 nor version 1.8.20  I see the section "Topic_subtopic", so
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by gbruti** [comment]
@albert-github I was able to reproduce the issue using files mentioned by @rolegic. I did one change, renamed those from `txt` to `cpp` (i think that 1.8.16, doesn't need this step).

`git clone ....doxygen.....
git checkout Release_1_8_16
cmake .
make
sudo make install`

I have put both files in directory "dox", here are the steps:

`
 [17:04:25]  test  cd dox                                                                                                                                                                                                                              
 [17:04:28]  dox  doxygen -g test                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Configuration file 'test' created.                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                               
Now edit the configuration file and enter                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                               
  doxygen test                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                               
to generate the documentation for your project                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                               
 [17:04:41]  dox  doxygen test                                                                                                                                                                                                                         
....
Building page list...
/export/mysql/test/dox/file2.cpp:2: warning: multiple use of page label 'my_doc', (other occurrence: /export/mysql/test/dox/file1.cpp, line: 3)
...
Generating example index...
finalizing index lists...
writing tag file...
Running plantuml with JAVA...
lookup cache used 0/65536 hits=0 misses=0
finished...

[17:04:41]  dox `

**Comment by gbruti** [comment]
I have just ran `git bisect` at Release_1_8_15 vs Release_1_8_16. It looks like this was intentional change done in:
```
commit cf4617ee1dc5b25b3e063863e886ac79a78d475c (refs/bisect/bad)
Author: albert-github <albert.tests@gmail.com>
Date:   Wed Aug 7 15:12:34 2019 +0200

    issue #7147: No warning when same page created
    
    Creating warning when adding documentation to an already existing page.
```

I am not an expert in regards doxygen, still shouldn't it apply only when we use same "page" multiple times and at least two usages have a title ?

**Comment by albert-github** [comment]
In general I think it is quite sensible to have this warning.
Normally doxygen will generate the documentation in a certain order of the file names, but it is nowhere written in which order. Furthermore when using a multi-threaded application (and doxygen has already some possibilities for it) it is not guaranteed anymore in which order the files will be processed and this in case of multiple runs the output might be different (different order of the blocks from the different files) and the user will be wondering why when he doesn't get a warning.

**Comment by albert-github** [comment]
@gbruti 
In your comment: https://github.com/doxygen/doxygen/issues/7471#issuecomment-744542219 you wrote:
> Now edit the configuration file and enter

what did you edit?

**Comment by gbruti** [comment]
@albert-github this is output of `doxygen -g test`. Maybe the example is not so clear that it was copied from terminal.

**Comment by doxygen** [comment]
I've made the warning more specific. Only case the same page label is used with a different, non-empty title, the warning is produced. The ordering issue mentioned by Albert is not really applicable. Doxygen can process multiple files in parallel these days, but it still combines the results in the input order which you can fully control if you mention the individual files in the configuration file explicitly.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
